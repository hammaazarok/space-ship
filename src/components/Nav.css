.nav {
  display: flex;
  justify-content: space-between;
  padding: 0 2rem 2rem 2rem;
}

.logo {
  width: 10vw;
  height: 10vw;
  padding-right: 1vw;
}

.left-nav {
  display: flex;
  align-items: center;
}

.left-nav h1 {
  font-size: 2vw;
  font-weight: bolder;
  color: gray;
}

.right-nav {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.nav:nth-child(4) {
  border-left: 2px solid grey;
}

.right-nav a {
  text-decoration: none;
  font-size: 2vw;
}

@media only screen and (max-width: 540px) {
  .nav {
    display: flex;
    justify-content: space-between;
    padding: 0.5rem;
  }

  .logo {
    width: 8vw;
    height: 8vw;
    padding-right: 0.5vw;
  }

  .left-nav h1 {
    font-size: 3vw;
  }

  .right-nav a {
    text-decoration: none;
    font-size: 3.3vw;
    gap: 0.3rem;
  }
}
