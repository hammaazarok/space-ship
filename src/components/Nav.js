import React from 'react';
import { NavLink } from 'react-router-dom';
import space from '../assets/images/planet2.png';
import './Nav.css';

const links = [
  {
    id: 1,
    path: '/',
    text: 'Rockets',
  },
  {
    id: 2,
    path: '/missions',
    text: 'Missions',
  },
  {
    id: 3,
    path: '/Profile',
    text: 'My profile',
  },

];

const Nav = () => (
  <div className="nav">
    <div className="left-nav">
      <img className="logo" src={space} alt="space" />
      <h1>Space Traveler&apos;s Hub</h1>
    </div>
    <div className="right-nav">
      {links.map((link) => (
        <NavLink
          className={(navlink) => (navlink.isActive ? 'active-nav' : '')}
          to={link.path}
          key={link.id}
        >
          {link.text}
        </NavLink>
      ))}
    </div>
  </div>
);

export default Nav;
