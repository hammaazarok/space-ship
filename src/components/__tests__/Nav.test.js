import React from 'react';
import Renderer from 'react-test-renderer';
import Nav from '../Nav';

it('renders correctly', () => {
  const tree = Renderer
    .create(<Nav />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
