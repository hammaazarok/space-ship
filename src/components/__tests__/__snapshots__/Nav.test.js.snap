// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav"
>
  <div
    className="left-nav"
  >
    <img
      alt="space"
      className="logo"
      src="planet2.png"
    />
    <h1>
      Space Traveler's Hub
    </h1>
  </div>
  <div
    className="right-nav"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "#00d7c6",
          "textDecoration": "underline",
        }
      }
    >
      Rockets
    </a>
    <a
      className=""
      href="/missions"
      onClick={[Function]}
      style={
        Object {
          "color": "RGB(246, 141, 54)",
          "textDecoration": "none",
        }
      }
    >
      Missions
    </a>
    <a
      className=""
      href="/Profile"
      onClick={[Function]}
      style={
        Object {
          "color": "RGB(246, 141, 54)",
          "textDecoration": "none",
        }
      }
    >
      My profile
    </a>
  </div>
</div>
`;
